app/javascript/mastodon/features/compose/components/compose_form.js

```diff
@@ -211,13 +212,14 @@ class ComposeForm extends ImmutablePureComponent {
     const disabled = this.props.isSubmitting;
 
     let publishText = '';
+    let publishButtonText = publish_button_text || intl.formatMessage(messages.publish);
```

Comment:
This is not great from localisation standpoint. You probably should check if the current locale is default instance locale and only then override the text, otherwise using the original message.

Reply:
I don't think this is necessarily advised. Since it would change what the button says depending on your locale. e.g in English you would see "Toot" but in German you would see "Veröffentlichen" ("Publish") which is inconsistent. In this case you would want to translate it into all languages but that doesn't require a setting, just a patch to the translation files. I think this setting should override any language preferences especially if its to theme the instance. A cat related instance should always say meow if the instance is primarily English (or whatever other word for non english instances). Users joining that instance would understand that its both a primarily English instance (no localization for the About section either) and that the publish button would match its theme and primary language